lib/open_food_network/reports/line_items.rb

```diff
@@ -51,13 +51,14 @@ def search_orders
         @order_permissions.visible_orders.complete.not_state(:canceled).search(@params[:q])
       end
 
-      def line_items_with_hidden_details(line_items)
-        editable_line_items = @order_permissions.editable_line_items.pluck(:id)
+      # From the line_items given, returns the ones that are editable by the user
+      def editable_line_items(line_items)
+        editable_line_items_ids = @order_permissions.editable_line_items.select(:id)
 
-        if editable_line_items.empty?
-          line_items
+        if editable_line_items_ids.empty?
+          Spree::LineItem.where("1=0")
         else
-          line_items.where('"spree_line_items"."id" NOT IN (?)', editable_line_items)
+          editable_line_items_ids.merge(line_items)
```

Comment:
Do we still need the if-else branch since we are not negating any more?

Reply:
:+1: removed, only less query!